File: assets/Siminov/Model/AdapterDescriptor.js
/**
	Exposes methods to GET and SET Hybrid Descriptor Adapter information as per define in HybridDescriptor.si.xml file or standalone xml file in application.
			
	Example:

		<adapter>
			<property name="name">name_of_adapter</property>
			<property name="description">description_about_adapter</property>
			<property name="type">type_of_adapter</property>
			<property name="map_to">map_to_class_name</property>
			<property name="cache">should_be_cached_or_not</property>
			
			<handlers>
			
				<handler>
					<property name="name">name_of_handler</property>
					<property name="description">description_about_handler</property>
					<property name="map_to">map_to_function_name</property>
					
					<parameters>
						
						<parameter>
							<property name="name"></property>
							<property name="description"></property>
							<property name="type"></property>
						</parameter>
					
					</parameter>
					
					<return>
							<property name="type">return_data_type</property>
							<property name="description">description_about_return_data</property>
					</return>
					
				</handler>
			
			</handlers>
			
		</adapter>

	@class HybridDescriptor.Adapter
	@constructor
				
*/
function AdapterDescriptor() {

    var properties = new Dictionary();

    var handlers = new Array();
    
	
	/**
	 * Get Name of Adapter.
	 * @return Name of Adapter.
	 */
    this.getName = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_NAME);
	}
	
	/**
	 * Set Name of Adapter.
	 * @param name Name of Adapter.
	 */
    this.setName = function(name) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_NAME, name);
	}

	/**
	 * Get Description of Adapter.
	 * @return Description of Adapter.
	 */
    this.getDescription = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_DESCRIPTION);
	}
	
	/**
	 * Set Description of Adapter.
	 * @param description Description of Adapter.
	 */
    this.setDescription = function(description) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_DESCRIPTION, description);
	}

	/**
	 * Get Type of Adapter.
	 * @return Type of Adapter.
	 */
    this.getType = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_TYPE);
	}
	
	/**
	 * Set Type of Adapter.
	 * @param type Type of Adapter.
	 */
    this.setType = function(type) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_TYPE, type);
	}

	/**
	 * Get Map To Name.
	 * @return Map To Name.
	 */
    this.getMapTo = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_MAP_TO);
	}
	
	/**
	 * Set Map To Name.
	 * @param mapTo Map To Name.
	 */
    this.setMapTo = function(mapTo) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_MAP_TO, mapTo);
	}

	/**
	 * Check whether cache is enabled or disabled.
	 * @return true/false; TRUE if cache enabled, FALSE if cache disabled.
	 */
    this.isCache = function() {
        return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_CACHE);
	}

	/**
	 * Set Cache value.
	 * @param cache Cache Enabled or Disabled.
	 */
    this.setCache = function(cache) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_CACHE, cache);
	}
	

	/**
	 	Get All Handler defined in descriptors.
	 
	 	@method getHandlers
		@return {Array} All Handlers.
	 */
    this.getHandlers = function() {
    	return handlers;
	}
	
	/**
	 	Add Handler.
	 	
	 	@method getHandlers
	 	@param handler {Handler} Handler.
	 */
    this.addHandler = function(handler) {
    	handlers.push(handler);
	}

	/**
	 	Get all Properties defined in descriptor.
	 
	 	@method getProperties
		@return {Array} All Property Values.
	 */
	this.getProperties = function() {
		return properties.values();
	}
	
	/**
	 	Get Property based on name provided.
	 
	 	@method getProperty
		@param name {String} Name of Property.
	 	@return {String} Property value.
	 */
	this.getProperty = function(name) {
		return properties.get(name);
	}
	
	/**
	 	Check whether Property exist or not.
	 
	 	@method containProperty
		@param name {String} Name of Property.
	 	@return {Boolean} true/false, TRUE if property exist, FALSE if property does not exist.
	 */
	this.containProperty = function(name) {
		return properties.exists(name);
	}
	
	/**
	 	Add Property in property pool.
	 
	 	@method addProperty
		@param name {String} Name of Property.
	 	@param value {String} value of Property.
	 */
	this.addProperty = function(name, value) {
		properties.add(name, value);
	}
	
	/**
	 	Remove Property from property pool.
	 
	 	@method removeProperty
		@param name {String} Name of Property.
	 */
	this.removeProperty = function(name) {
		properties.remove(name);
	}
	
    
} 


/**
 	Exposes methods to GET and SET Hybrid Descriptor Adapter Handler information as per define in HybridDescriptor.si.xml file or in standalone adapter xml file in application.

	@class HybridDescriptor.Adapter.Handler
 */
AdapterDescriptor.Handler = function() {

    var properties = new Dictionary();

    var parameters = new Array();


	/**
	 	Get Name of Handler.
	 	
	 	@method getName
	 	@return {String} Name of Handler. 
	 */
    this.getName = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_HANDLER_NAME);
	}
	
	/**
	 	Set Name of Handler.
	 	
	 	@method setName
	 	@param name {String} Name of handler.
	 */
    this.setName = function(name) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_HANDLER_NAME, name);
	}


	/**
	 	Get Description about Handler.
	 
	 	@method getDescription
		@return {String} Description about Handler.
	 */
    this.getDescription = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_HANDLER_DESCRIPTION);
	}
	
	/**
	 	Set Description about Handler.
	 
	 	@method setDescription
		@param description {String} Description about Handler.
	 */
    this.setDescription = function(description) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_HANDLER_DESCRIPTION, description);
	}

	/**
	 	Get Map To Name.
	 
	 	@method getMapTo
		@return {String} Map To Name.
	 */
    this.getMapTo = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_HANDLER_MAP_TO);
	}
	
	/**
	 	Set Map To Name.
	 
	 	@method setMapTo
		@param mapTo {String} Map To Name.
	 */
    this.setMapTo = function(mapTo) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_HANDLER_MAP_TO, mapTo);
	}


	/**
	 	Get all Properties defined in descriptor.
	 
	 	@method getProperties
		@return {Array} All Property Values.
	 */
	this.getProperties = function() {
		return properties.values();
	}
	
	/**
	 	Get Property based on name provided.
	 
	 	@method getProperty
		@param name {String} Name of Property.
	 	@return {String} Property value.
	 */
	this.getProperty = function(name) {
		return properties.get(name);
	}
	
	/**
	 	Check whether Property exist or not.
	 
	 	@method containProperty
		@param name {String} Name of Property.
	 	@return {Boolean} true/false, TRUE if property exist, FALSE if property does not exist.
	 */
	this.containProperty = function(name) {
		return properties.exists(name);
	}
	
	/**
	 	Add Property in property pool.
	 
	 	@method addProperty
		@param name {String} Name of Property.
	 	@param value {String} value of Property.
	 */
	this.addProperty = function(name, value) {
		properties.add(name, value);
	}
	
	/**
	 	Remove Property from property pool.
	 
	 	@method removeProperty
		@param name {String} Name of Property.
	 */
	this.removeProperty = function(name) {
		properties.remove(name);
	}
	

} 


/**
 	Exposes methods to GET and SET Parameter information as per define in HybridDescriptor.si.xml file or in standalone adapter xml file in application.
 	
 	@class HybridDescription.Adapter.Handler.Parameter
 */
AdapterDescriptor.Handler.Parameter = function() {

    var properties = new Dictionary();

	/**
	 	Get Name of Parameter.
	 	
	 	@method getDescription
	 	@return {String} Name of Parameter.
	 */
    this.getDescription = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_PARAMETER_DESCRIPTION);
	}
	
	/**
	 	Set Name of Parameter.
	 	
	 	@method setDescription
	 	@param name {String} Name of Parameter.
	 */
    this.setDescription = function(description) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_PARAMETER_DESCRIPTION, description);
	}

	/**
	 	Get Type of Parameter.
	 	
	 	@method getType
	 	@return {String} Type of Parameter.
	 */
    this.getType = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_PARAMETER_TYPE);
	}
	
	/**
	 	Get Type of Parameter.
	 	
	 	@method setType
	 	@param type {String} Type of Parameter.
	 */
    this.setType = function(type) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_PARAMETER_TYPE, type);
	}


	/**
	 	Get all Properties defined in descriptor.
	 
	 	@method getProperties
		@return {Array} All Property Values.
	 */
	this.getProperties = function() {
		return properties.values();
	}
	
	/**
	 	Get Property based on name provided.
	 
	 	@method getProperty
		@param name {String} Name of Property.
	 	@return {String} Property value.
	 */
	this.getProperty = function(name) {
		return properties.get(name);
	}
	
	/**
	 	Check whether Property exist or not.
	 
	 	@method containProperty
		@param name {String} Name of Property.
	 	@return {Boolean} true/false, TRUE if property exist, FALSE if property does not exist.
	 */
	this.containProperty = function(name) {
		return properties.exists(name);
	}
	
	/**
	 	Add Property in property pool.
	 
	 	@method addProperty
		@param name {String} Name of Property.
	 	@param value {String} value of Property.
	 */
	this.addProperty = function(name, value) {
		properties.add(name, value);
	}
	
	/**
	 	Remove Property from property pool.
	 
	 	@method removeProperty
		@param name {String} Name of Property.
	 */
	this.removeProperty = function(name) {
		properties.remove(name);
	}

} 


/**
 	Exposes methods to GET and SET Return information as per define in HybridDescriptor.si.xml file or in standalone adapter xml file in application.
 */
AdapterDescriptor.Handler.Return = function() {

    var properties = new Dictionary();


	/**
	 	Get Description about Return.
	 	
	 	@method getDescription
	 	@return {String} Description about Return.
	 */
    this.getDescription = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_RETURN_DESCRIPTION);
	}
	
	/**
	 	Set Description about Return.
	 	
	 	@method setDescription
	 	@param description {String} Description about Return.
	 */
    this.setDescription = function(description) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_RETURN_DESCRIPTION, description);
	}

	/**
	 	Get Type of Return.
	 	
	 	@method getType
	 	@return {String} Type of Return.
	 */
    this.getType = function() {
    	return properties.get(Constants.HYBRID_DESCRIPTOR_ADAPTER_RETURN_TYPE);
	}
	
	/**
	 	Set Type of Return.
	 	
	 	@method setType
	 	@param type {String} Type of Return.
	 */
    this.setType = function(type) {
    	properties.add(Constants.HYBRID_DESCRIPTOR_ADAPTER_RETURN_TYPE, type);
	}

	/**
	 	Get all Properties defined in descriptor.
	 
	 	@method getProperties
		@return {Array} All Property Values.
	 */
	this.getProperties = function() {
		return properties.values();
	}
	
	/**
	 	Get Property based on name provided.
	 
	 	@method getProperty
		@param name {String} Name of Property.
	 	@return {String} Property value.
	 */
	this.getProperty = function(name) {
		return properties.get(name);
	}
	
	/**
	 	Check whether Property exist or not.
	 
	 	@method containProperty
		@param name {String} Name of Property.
	 	@return {Boolean} true/false, TRUE if property exist, FALSE if property does not exist.
	 */
	this.containProperty = function(name) {
		return properties.exists(name);
	}
	
	/**
	 	Add Property in property pool.
	 
	 	@method addProperty
		@param name {String} Name of Property.
	 	@param value {String} value of Property.
	 */
	this.addProperty = function(name, value) {
		properties.add(name, value);
	}
	
	/**
	 	Remove Property from property pool.
	 
	 	@method removeProperty
		@param name {String} Name of Property.
	 */
	this.removeProperty = function(name) {
		properties.remove(name);
	}
} 

File: src/siminov/hybrid/model/AdapterDescriptor.java
/** 
 * [SIMINOV FRAMEWORK]
 * Copyright [2015] [Siminov Software Solution LLP|support@siminov.com]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 **/

package siminov.hybrid.model;

import java.util.Collection;
import java.util.HashMap;
import java.util.Iterator;
import java.util.LinkedList;
import java.util.Map;

import siminov.hybrid.Constants;
import siminov.orm.model.IDescriptor;


/**
 * Exposes methods to GET and SET Hybrid Descriptor Adapter information as per define in HybridDescriptor.si.xml file or standalone xml file in application.
	<p>
		<pre>
		
Example:
	{@code
	<adapter>
		<property name="name">name_of_adapter</property>
		<property name="description">description_about_adapter</property>
		<property name="type">type_of_adapter</property>
		<property name="map_to">map_to_class_name</property>
		<property name="cache">should_be_cached_or_not</property>
		
		<handlers>
		
			<handler>
				<property name="name">name_of_handler</property>
				<property name="description">description_about_handler</property>
				<property name="map_to">map_to_function_name</property>
				
				<parameters>
					
					<parameter>
						<property name="name"></property>
						<property name="description"></property>
						<property name="type"></property>
					</parameter>
				
				</parameter>
				
				<return>
						<property name="type">return_data_type</property>
						<property name="description">description_about_return_data</property>
				</return>
				
			</handler>
		
		</handlers>
		
	</adapter>
			
	}
	
		</pre>
	</p>
 */
public class AdapterDescriptor implements IDescriptor {

	private Map<String, String> properties = new HashMap<String, String>();
	
	private Map<String, Handler> handlers = new HashMap<String, Handler>();
	
	/**
	 * Get Name of Adapter.
	 * @return Name of Adapter.
	 */
	public String getName() {
		return this.properties.get(Constants.ADAPTER_DESCRIPTOR_PROPERTY_NAME);
	}
	
	/**
	 * Set Name of Adapter.
	 * @param name Name of Adapter.
	 */
	public void setName(final String name) {
		this.properties.put(Constants.ADAPTER_DESCRIPTOR_PROPERTY_NAME, name);
	}
	
	/**
	 * Get Description of Adapter.
	 * @return Description of Adapter.
	 */
	public String getDescription() {
		return this.properties.get(Constants.ADAPTER_DESCRIPTOR_PROPERTY_DESCRIPTION);
	}
	
	/**
	 * Set Description of Adapter.
	 * @param description Description of Adapter.
	 */
	public void setDescription(final String description) {
		this.properties.put(Constants.ADAPTER_DESCRIPTOR_PROPERTY_DESCRIPTION, description);
	}
	
	/**
	 * Get Type of Adapter.
	 * @return Type of Adapter.
	 */
	public String getType() {
		return this.properties.get(Constants.ADAPTER_DESCRIPTOR_PROPERTY_TYPE);
	}
	
	/**
	 * Set Type of Adapter.
	 * @param type Type of Adapter.
	 */
	public void setType(final String type) {
		this.properties.put(Constants.ADAPTER_DESCRIPTOR_PROPERTY_TYPE, type);
	}
	
	/**
	 * Get Map To Name.
	 * @return Map To Name.
	 */
	public String getMapTo() {
		return this.properties.get(Constants.ADAPTER_DESCRIPTOR_PROPERTY_MAP_TO);
	}
	
	/**
	 * Set Map To Name.
	 * @param mapTo Map To Name.
	 */
	public void setMapTo(final String mapTo) {
		this.properties.put(Constants.ADAPTER_DESCRIPTOR_PROPERTY_MAP_TO, mapTo);
	}

	/**
	 * Check whether cache is enabled or disabled.
	 * @return true/false; TRUE if cache enabled, FALSE if cache disabled.
	 */
	public boolean isCache() {
		String cache = this.properties.get(Constants.ADAPTER_DESCRIPTOR_PROPERTY_CACHE);
		
		if(cache != null && cache.length() > 0 && cache.equalsIgnoreCase("true")) {
			return true;
		}
		
		return false;
	}
	
	/**
	 * Set Cache value.
	 * @param cache Cache Enabled or Disabled.
	 */
	public void setCache(final boolean cache) {
		this.properties.get(Constants.ADAPTER_DESCRIPTOR_PROPERTY_CACHE);
	}

	/**
	 * Get all Properties defined in descriptor.
	 * @return All Property Values.
	 */
	public Iterator<String> getProperties() {
		return this.properties.keySet().iterator();
	}
	
	/**
	 * Get Property based on name provided.
	 * @param name Name of Property.
	 * @return Property value.
	 */
	public String getProperty(String name) {
		return this.properties.get(name);
	}

	/**
	 * Check whether Property exist or not.
	 * @param name Name of Property.
	 * @return true/false, TRUE if property exist, FALSE if property does not exist.
	 */
	public boolean containProperty(String name) {
		return this.properties.containsKey(name);
	}
	
	/**
	 * Add Property in property pool.
	 * @param name Name of Property.
	 * @param value value of Property.
	 */
	public void addProperty(String name, String value) {
		this.properties.put(name, value);
	}
	
	/**
	 * Remove Property from property pool.
	 * @param name Name of Property.
	 */
	public void removeProperty(String name) {
		this.properties.remove(name);
	}

	
	/**
	 * Get All Handler defined in descriptors.
	 * @return All Handlers.
	 */
	public Iterator<Handler> getHandlers() {
		return this.handlers.values().iterator();
	}
	
	/**
	 * Get Handler based on handler name.
	 * @param handlerName Name of Handler.
	 * @return Handler.
	 */
	public Handler getHandler(final String handlerName) {
		return this.handlers.get(handlerName);
	}
	
	/**
	 * Add Handler.
	 * @param handler Handler.
	 */
	public void addHandler(final Handler handler) {
		this.handlers.put(handler.getName(), handler);
	}
	
	/**
	 * Check whether handler exist or not based on handler name.
	 * @param handlerName Name of handler.
	 * @return
	 */
	public boolean containHandler(final String handlerName) {
		return this.handlers.containsKey(handlerName);
	}

	/**
	 * Exposes methods to GET and SET Hybrid Descriptor Adapter Handler information as per define in HybridDescriptor.si.xml file or in standalone adapter xml file in application.
	 *
	 */
	public static class Handler implements IDescriptor {

		private Map<String, String> properties = new HashMap<String, String>();
		
		private Collection<Parameter> parameters = new LinkedList<Parameter>();
		private Return returnData = null;

		/**
		 * Get Name of Handler.
		 * @return Name of Handler. 
		 */
		public String getName() {
			return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_PROPERTY_NAME);
		}
		
		/**
		 * Set Name of Handler.
		 * @param name Name of handler.
		 */
		public void setName(final String name) {
			this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_PROPERTY_NAME, name);
		}
		
		/**
		 * Get Map To Name.
		 * @return Map To Name.
		 */
		public String getMapTo() {
			return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_PROPERTY_MAP_TO);
		}
		
		/**
		 * Set Map To Name.
		 * @param mapTo Map To Name.
		 */
		public void setMapTo(final String mapTo) {
			this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_PROPERTY_MAP_TO, mapTo);
		}
		
		/**
		 * Get Description about Handler.
		 * @return Description about Handler.
		 */
		public String getDescription() {
			return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_PROPERTY_DESCRIPTION);
		}
		
		/**
		 * Set Description about Handler.
		 * @param description Description about Handler.
		 */
		public void setDescription(final String description) {
			this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_PROPERTY_DESCRIPTION, description);
		}
		
		/**
		 * Get all Properties defined in descriptor.
		 * @return All Property Values.
		 */
		public Iterator<String> getProperties() {
			return this.properties.keySet().iterator();
		}
		
		/**
		 * Get Property based on name provided.
		 * @param name Name of Property.
		 * @return Property value.
		 */
		public String getProperty(String name) {
			return this.properties.get(name);
		}

		/**
		 * Check whether Property exist or not.
		 * @param name Name of Property.
		 * @return true/false, TRUE if property exist, FALSE if property does not exist.
		 */
		public boolean containProperty(String name) {
			return this.properties.containsKey(name);
		}
		
		/**
		 * Add Property in property pool.
		 * @param name Name of Property.
		 * @param value value of Property.
		 */
		public void addProperty(String name, String value) {
			this.properties.put(name, value);
		}
		
		/**
		 * Remove Property from property pool.
		 * @param name Name of Property.
		 */
		public void removeProperty(String name) {
			this.properties.remove(name);
		}

		/**
		 * Get All Parameters defined.
		 * @return All Parameters.
		 */
		public Iterator<Parameter> getParameters() {
			return this.parameters.iterator();
		}
		
		/**
		 * Add Parameter.
		 * @param parameter Parameter.
		 */
		public void addParameter(final Parameter parameter) {
			this.parameters.add(parameter);
		}
		
		/**
		 * Get Return Object of Handler.
		 * @return Return Object.
		 */
		public Return getReturn() {
			return this.returnData;
		}
		
		/**
		 * Set Return Object of Handler.
		 * @param returnData Return Object.
		 */
		public void setReturn(final Return returnData) {
			this.returnData = returnData;
		}
		
		/**
		 * Exposes methods to GET and SET Parameter information as per define in HybridDescriptor.si.xml file or in standalone adapter xml file in application.
		 *
		 */
		public static class Parameter implements IDescriptor {

			private Map<String, String> properties = new HashMap<String, String>();

			/**
			 * Get Name of Parameter.
			 * @return Name of Parameter.
			 */
			public String getName() {
				return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_PARAMETER_PROPERTY_NAME);
			}
			
			/**
			 * Set Name of Parameter.
			 * @param name Name of Parameter.
			 */
			public void setName(final String name) {
				this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_PARAMETER_PROPERTY_NAME, name);
			}
			
			/**
			 * Get Type of Parameter.
			 * @return Type of Parameter.
			 */
			public String getType() {
				return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_PARAMETER_PROPERTY_TYPE);
			}
			
			/**
			 * Get Type of Parameter.
			 * @param type Type of Parameter.
			 */
			public void setType(final String type) {
				this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_PARAMETER_PROPERTY_TYPE, type);
			}
			
			/**
			 * Get Description about Parameter.
			 * @return Description about Parameter.
			 */
			public String getDescription() {
				return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_PARAMETER_PROPERTY_DESCRIPTION);
			}
			
			/**
			 * Set Description about Parameter.
			 * @param description Description about Parameter.
			 */
			public void setDescription(final String description) {
				this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_PARAMETER_PROPERTY_DESCRIPTION, description);
			}
			
			/**
			 * Get all Properties defined in descriptor.
			 * @return All Property Values.
			 */
			public Iterator<String> getProperties() {
				return this.properties.keySet().iterator();
			}
			
			/**
			 * Get Property based on name provided.
			 * @param name Name of Property.
			 * @return Property value.
			 */
			public String getProperty(String name) {
				return this.properties.get(name);
			}

			/**
			 * Check whether Property exist or not.
			 * @param name Name of Property.
			 * @return true/false, TRUE if property exist, FALSE if property does not exist.
			 */
			public boolean containProperty(String name) {
				return this.properties.containsKey(name);
			}
			
			/**
			 * Add Property in property pool.
			 * @param name Name of Property.
			 * @param value value of Property.
			 */
			public void addProperty(String name, String value) {
				this.properties.put(name, value);
			}
			
			/**
			 * Remove Property from property pool.
			 * @param name Name of Property.
			 */
			public void removeProperty(String name) {
				this.properties.remove(name);
			}

		}

		
		
		/**
		 * Exposes methods to GET and SET Return information as per define in HybridDescriptor.si.xml file or in standalone adapter xml file in application.
		 *
		 */
		public static class Return implements IDescriptor {
			
			private Map<String, String> properties = new HashMap<String, String>();

			/**
			 * Get Type of Return.
			 * @return Type of Return.
			 */
			public String getType() {
				return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_RETURN_PROPERTY_TYPE);
			}
			
			/**
			 * Set Type of Return.
			 * @param type Type of Return.
			 */
			public void setType(final String type) {
				this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_RETURN_PROPERTY_TYPE, type);
			}
			
			/**
			 * Get Description about Return.
			 * @return Description about Return.
			 */
			public String getDescription() {
				return this.properties.get(Constants.ADAPTER_DESCRIPTOR_HANDLER_RETURN_PROPERTY_DESCRIPTION);
			}
			
			/**
			 * Set Description about Return.
			 * @param description Description about Return.
			 */
			public void setDescription(final String description) {
				this.properties.put(Constants.ADAPTER_DESCRIPTOR_HANDLER_RETURN_PROPERTY_DESCRIPTION, description);
			}
			
			/**
			 * Get all Properties defined in descriptor.
			 * @return All Property Values.
			 */
			public Iterator<String> getProperties() {
				return this.properties.keySet().iterator();
			}
			
			/**
			 * Get Property based on name provided.
			 * @param name Name of Property.
			 * @return Property value.
			 */
			public String getProperty(String name) {
				return this.properties.get(name);
			}

			/**
			 * Check whether Property exist or not.
			 * @param name Name of Property.
			 * @return true/false, TRUE if property exist, FALSE if property does not exist.
			 */
			public boolean containProperty(String name) {
				return this.properties.containsKey(name);
			}
			
			/**
			 * Add Property in property pool.
			 * @param name Name of Property.
			 * @param value value of Property.
			 */
			public void addProperty(String name, String value) {
				this.properties.put(name, value);
			}
			
			/**
			 * Remove Property from property pool.
			 * @param name Name of Property.
			 */
			public void removeProperty(String name) {
				this.properties.remove(name);
			}

		}
	}
}
